src/annotations/AnnotationManager.php

```diff
@@ -322,6 +346,25 @@ protected function getAnnotations($class_name, $member_type = self::MEMBER_CLASS
     }
 
     /**
+     * Determines whether a class or trait has the specified member.
+     *
+     * @param string $className The name of the class or trait to check
+     * @param string $memberType The type of member, e.g. "property" or "method"
+     * @param string $memberName The member name, e.g. "method" or "$property"
+     *
+     * @return bool whether class or trait has the specified member
+     */
+    protected function classHasMember($className, $memberType, $memberName)
+    {
+        if ($memberType === self::MEMBER_METHOD) {
+            return method_exists($className, $memberName);
+        } else if ($memberType === self::MEMBER_PROPERTY) {
+            return property_exists($className, ltrim($memberName, '$'));
+        }
+        return false;
+    }
```

Comment:
I've just tried that myself and some unrelated tests started to fail. Probably ones, that expected inherited annotations to kick in even if property is defined in sub-class.

Let's keep your current solution for now.